src/compiler/checker.ts

```diff
@@ -24255,7 +24267,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 if (isFromInferenceBlockedSource(source)) {
                     return;
                 }
-                const inference = getInferenceInfoForType(target);
+                let inference = getInferenceInfoForType(target);
+                if (
+                    !inference && (target.flags & TypeFlags.IndexedAccess) &&
+                    inferences[0].typeParameter.flags & TypeFlags.IndexedAccess &&
+                    (target as IndexedAccessType).indexType.flags & TypeFlags.StringLiteral &&
+                    ((target as IndexedAccessType).objectType.flags & TypeFlags.IndexedAccess)
+                ) {
+                    inferences.push(inference = createInferenceInfo(target));

```

Comment:
I tried to learn from #52944 but that PR is quite a lot to take in for me 😅 So I decided to reuse ur work from #20126 as it was much simpler to pull off. 

After some tweaks I managed to make it work for the reverse mapped types case that this PR originally was meant to fix. There are currently some issues with the current state of this PR:
- some of the tests that you originally added were changed, you can see the commit that updates those baselines [here](https://github.com/microsoft/TypeScript/pull/53017/commits/d0e8b8b4b9e49447921d7c1646c92488bccb66bf) (some are just display changes, some are not though)
- self-check fails

I intend to investigate both problems but perhaps you'd be able to peek here in the meantime and validate the used approach. It would be great to know if I'm heading into the right direction here.